sqlmesh/core/audit/builtin.py

```diff
@@ -60,3 +60,16 @@
 HAVING COUNT(*) <= @threshold
     """,
 )
+
+
+forall_audit = Audit(
+    name="forall",
+    query="""
+SELECT *
+FROM @this_model
+WHERE @REDUCE(
+  @criteria,
+  (l, r) -> NOT(@SQL(l)) OR NOT(@SQL(r))
```

Comment:
is the @SQL necessary here? aren't l and r ast nodes?

with the NOT let's add spaces because it looks like a function call but it's not

NOT (@l) OR NOT (@r)

Reply:
I wasn't sure if `@SQL` was necessary. I had some errors at some point and this seemed to fix it, let me try without it.

Reply:
yep, `@SQL` wasn't necessary, fixed in https://github.com/TobikoData/sqlmesh/pull/698/commits/eccc37f0ed90ae19a64f265912de33d71c105af9